apps/trustlab/src/lib/data/common/index.js

```diff
@@ -0,0 +1,131 @@
+function getPageSlug({ params }) {
+  const slugsCount = params?.slugs?.length;
+  // count < 3, page slug is the last slug e.g. ["about"] or ["knowldge/news"]
+  // count == 3, page slug is the 2nd slug (index 1); last slug (index 3)
+  //             is the post. e.g. opportunities/grants/democratic-governance-in-zambia
+  const pageSlugIndex = slugsCount < 3 ? slugsCount - 1 : 1;
+  return params?.slugs?.[pageSlugIndex] || "index";
+}
+
+function getDefaultErrorPageProps(slug = "404") {
+  if (slug === "500") {
+    return {
+      blocks: [
```

Comment:
I'm assuming we're going to move these into PayloadCMS, right?

Reply:
@kilemensi We will have a 404 page and component in the CMS. Next.js, however, builds the error pages at [build time](https://nextjs.org/docs/pages/building-your-application/routing/custom-error#customizing-the-404-page)

> To create a custom 404 page, you can create a pages/404.js file. This file is statically generated at build time.

The default error props are for situations where the 404/500 page has not been created yet in the db at build time.